dspace-server-webapp/src/main/java/org/dspace/app/rest/model/UploadAccessConditionDTO.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.model;
+
+import java.util.Date;
+import java.util.UUID;
+
+/**
+ * The UploadAccessCondition it is partial representation of the DSpace ResourcePolicy
+ *
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk at 4science.it)
+ */
+public class UploadAccessConditionDTO  {
```

Comment:
I have to admit, this class is confusing to me. What is an "UploadAccessCondition" and what does it have to do with a ResourcePolicy?  Why do we require a "partial representation" of a ResourcePolicy? Why is this class a DTO, when we tend to create `*Rest` model objects for the REST API layer?

Also, I'm noticing we already have an `org.dspace.app.rest.model.AccessConditionOptionRest` object and an `org.dspace.app.rest.model.ResourcePolicyRest` object.  What is this class doing that couldn't be handled by one of those other existing objects? 

(To clarify here, I'm not against adding this DTO if it's needed. But, I'd like us to better describe *why* it is needed and how it differs from the previously mentioned Rest objects. It's currently not clear to me why we need this object.)

Reply:
yes, we need an additional class to represent the payload used by the rest client to update information in the upload section of a workspaceitem. I will add more information about it in the javadoc, but the main difference is that the `ResourcePolicyRest` object contains a reference to the real Group object but here we only need the uuid (the uri-list approach cannot be used inside the patch). The `AccessConditionOptionRest` instead is used to represent one option to set access in the upload pannel, such as open acces, embargo, leave, etc.

You are of course right about use this new class everywhere in the patch, we will look to fix that

Reply:
Thanks @abollini for the better JavaDocs here, as it's helped me to better wrap my head around how this is working.  I'll now do a more thorough review.